### PR TITLE
Render directly using FFMPEG instead of VideoIO

### PR DESCRIPTION
**Pull request by ArbitRandomUser**
## PR Checklist

If you are contributing to `Javis.jl`, please make sure you are able to check off each item on this list:

- [x] Did I update `CHANGELOG.md` with whatever changes/features I added with this PR?
- [x] Did I make sure to only change the part of the file where I introduced a new change/feature?
- [x] Did I cover all corner cases to be close to 100% test coverage (if applicable)?
- [x] Did I properly add Javis dependencies to the `Project.toml` + set an upper bound of the dependency (if applicable)?
- [x] Did I properly add test dependencies to the `test` directory (if applicable)?
- [x] Did I check relevant tutorials that may be affected by changes in this PR?
- [x] Did I clearly articulate why this PR was made the way it was and how it was made?

**Link to relevant issue(s)**
Closes #

**How did you address these issues with this PR? What methods did you use?**
This PR gets rid of  VideoIO altogether for video rendering , some testing showed significant memory usage with using VideoIO , indicative of memory leaks .
This PR aims to fix huge RAM usage on repeatedly rendering (typically happens when using Javis in the REPL iterating changes into a video )
FFMPEG is used directly to stitch the frames together.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ version = "0.9.0"
 Animations = "27a7e980-b3e6-11e9-2bcd-0b925532e340"
 Cairo = "159f3aea-2a34-519c-b102-8c37f9878175"
 FFMPEG = "c87230d0-a227-11e9-1b43-d7ebe4e7570a"
+FileIO = "5789e2e9-d7fb-5bc7-8068-2c6fae9b9549"
```
Comment (Wikunia) [inline Project.toml:10]: I think we can remove VideoIO here and have it as a test dependency only, right?

Comment (ArbitRandomUser) [reply Project.toml:10]: yes that should work .

File: src/Javis.jl [hunk 2]
```diff
@@ -309,6 +308,10 @@ function render(
     started_encoding_video = false
 
     filecounter = 1
+    binpath = joinpath(FFMPEG.FFMPEG_jll.PATH_list[end], "ffmpeg") #is ffmpeg bin always at the end ?
```
Comment (Wikunia) [inline src/Javis.jl:311]: can't we use `ffmpeg` like here: https://docs.juliahub.com/FFMPEG/TGvga/0.4.0/autodocs/#FFMPEG.@ffmpeg_env-Tuple%7BAny%7D

**Review by Wikunia — COMMENTED** [review]
Besides the one comment:
I have the feeling that the memory consumption is still rising when I run something like:
```
for i in 1:100
   include("animate.jl")
end
```
for any kind of animate.jl script that renders to an mp4. In my case running `GC.gc()` reduced it back to the previous level though.

File: src/Javis.jl (hunk 2 above)
Comment (ArbitRandomUser) [reply src/Javis.jl:311]: yes indeed ,

**Comment by ArbitRandomUser** [comment]
I did notice some memory rise on mine , but its definitely not as much as with VideoIO.jl .
it usually tends to plateau after some iterations of the loop .( and sometimes even drops down and rises again) . It might just be the GC deciding when to cleanup .

The important thing I  think the current memory is  "GC-able" . 
Unlike with  VideoIO.jl   where memory would not get cleared by the GC.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # Javis.jl - Changelog
 
+# PR changes
+- changed render method for mp4 to use ffmpeg directly inplace of VideoIO
+- TODO change test to use ffprobe for  video length and we can get rid of VideoIO completely
```
Comment (Wikunia) [inline CHANGELOG.md:5]: Can you add this todo in the code and remove it from the changelog?

File: src/Javis.jl [hunk 4]
```diff
@@ -752,4 +751,7 @@ export setup_stream, cancel_stream
 # scales
 export scale_linear, @scale_layer
 
+export morph
```
Comment (Wikunia) [inline src/Javis.jl:754]: can you remove these three export lines? Seems like they are from your other branches?

File: src/Javis.jl [hunk 5]
```diff
@@ -699,6 +697,7 @@ function set_object_defaults!(object)
     scaleto(desired_scale)
 end
 
+#todo add stroke/fills here
```
Comment (Wikunia) [inline src/Javis.jl:700]: Also this doesn't seem to be related to this PR.

**Review by Wikunia — COMMENTED** [review]
Will merge when those three things are done :wink: 
Thanks again for finding this out and coming up with this fix.

File: src/Javis.jl (hunk 5 above)
Comment (ArbitRandomUser) [reply src/Javis.jl]: remove

File: CHANGELOG.md (hunk 3 above)
Comment (ArbitRandomUser) [reply CHANGELOG.md]: done

**Comment by ArbitRandomUser** [comment]
Removed the one line for which we needed VideoIO as a testdep , and replaced it with a line using `ffprobe`.
So now we dont need VideoIO even as a test dependency.

**Comment by ArbitRandomUser** [comment]
okay thats strange ... the tests worked on my machine , but doesnt seem to be working on github.

**Comment by Wikunia** [comment]
I wouldn't worry about the test dependency.
